bazel/src/main/java/com/google/api/codegen/bazel/BazelBuildFileView.java

```diff
@@ -43,6 +44,10 @@ class BazelBuildFileView {
     tokens.put("version", bp.getVersion());
     tokens.put("package", bp.getProtoPackage());
 
+    // For regeneration of Java rules, we are particularly interested in what the saved transport value was,
+    // if there was one, in order to correctly generate, or not, the rest/grpc specific targets and labels.
+    String javaTransport = bp.getJavaTransportOverride() != null ? bp.getJavaTransportOverride() : transport;
```

Comment:
stylistic nit: I'd avoid calling the function twice, even if it means an extra line of code.